test/jdk/java/util/concurrent/Executors/UnreferencedExecutor.java

```diff
@@ -36,10 +36,33 @@ public class UnreferencedExecutor {
 
     private static final int DURATION_IN_SECONDS = 5;
 
+    /**
+     * An UncaughtExceptionHandler to ignore recoverable OOMEs
+     * caused by other VM threads that do allocation or create
+     * temporary strong roots to otherwise weakly referenced objects.
+     * This can happen for example when using libgraal and -Xcomp.
+     */
+    static class UEH implements Thread.UncaughtExceptionHandler {
+        @Override
+        public void uncaughtException(Thread t, Throwable e) {
+            if (e instanceof OutOfMemoryError err) {
+                // Ignore. An unrecoverable memory leak will still
+                // cause the VM to exit.
+            } else if (e instanceof Error err) {
+                throw err;
+            } else if (e instanceof RuntimeException re) {
+                throw re;
+            } else {
+                throw new RuntimeException(e);
+            }
+        }
```

Comment:
I'm not clear exactly what you are trying to achieve with this - is it just to not print the stacktrace when OOME occurs?

A UEH should not rethrow exceptions as they are effectively ignored as we are already at the final level of handling exceptions (the VM will print a one line warning if a UEH itself throws). If I understand what you want then I think the following would be more correct:
```
// Ignore OOME but do default handling for any other uncaught exception.
public void uncaughtException(Thread t, Throwable e) {
    if ( ! e instanceof OutOfMemoryError) {
        System.err.print("Exception in thread \"" + t.getName() + "\" ");
        e.printStackTrace(System.err);
    }
}
```

This doesn't stop the thread from terminating of course - the OOME already caused it to unwind its stack if we get this far.